specification/transport_layer/can.tex

```diff
@@ -261,23 +278,28 @@ \subsubsection{Payload discriminator}\label{sec:can_payload_discriminator}
 for different data field values.
 A possible way of initializing the payload discriminator value is to apply the transfer CRC function
 (as defined in the section \ref{sec:transfer_crc})
-to the contents of the anonymous message, and then use any 10 bits of the result.
+to the contents of the anonymous message, and then use the seven least significant bits of the result.
 Nodes that adopt this approach will be using the same payload discriminator value for identical messages,
 which is acceptable since this will not trigger an error on the bus.
 
-Since the discriminator is only 10 bits long,
-the probability of having multiple nodes emitting CAN frames with the same CAN ID
-but different data can exceed 0.1\%, which is significant.
+Since the discriminator is only seven bits long (128 possible values),
+a collision where multiple nodes emit CAN frames with different data but same CAN ID is likely to happen
+despite the randomization measures described earlier.
 Therefore, the protocol must account for possible errors on the CAN bus triggered by CAN ID collisions.
 In order to comply with this requirement,
 UAVCAN requires all nodes to immediately abort transmission of all anonymous transfers once an error on
 the CAN bus is detected.
 This measure allows the protocol to prevent the bus deadlock that may occur if the automatic
 retransmission on bus error is not suppressed.
+Applications should use some form of CSMA/CD\footnote{Carrier-sense multiple access with collision detection
```

Comment:
This is a very confusing statement to encounter given that CAN is a CSMA protocol. We should acknowledge that this is an application-level implementation of CSMA on top of a bus that does CSMA in hardware.

Reply:
Is it better to define a few rules?
 - An anonymous message that fails to be transmitted must not be retransmitted with the same discriminator unless a scheme for an indeterministic (guaranteed low correlation with other nodes on the bus) chosen to wait is used.
 - An anonymous message that fails to be transmitted must not be retransmitted immediately unless a scheme for indeterministic (guaranteed low correlation with other nodes on the bus) discriminator selection is chosen.

Reply:
It seems sufficient to say that we need CSMA/CD; spelling out the implementation details seems unnecessary.

Reply:
Moved CSMA/CD to the footnote.